src/bosh-google-cpi/action/attach_disk.go

```diff
@@ -38,26 +40,56 @@ func (ad AttachDisk) Run(vmCID VMCID, diskCID DiskCID) (interface{}, error) {
 		return nil, api.NewDiskNotFoundError(string(diskCID), false)
 	}
 
+	// If this disk isn't already attached to this VM at the IAAS level, do that now
+	if !diskAttachedToVM(vmCID, disk.Users) {
```

Comment:
If the disk is already attached, do we want to proceed with the logic below? If not, then it might be worthwhile reversing the conditional logic here and returning early.

Reply:
I initially structure it that way, but yeah, the [logic below](https://github.com/cloudfoundry/bosh-google-cpi-release/blob/1b71cf973d94b8b92f3e1dd86f7855f946d20bb5/src/bosh-google-cpi/action/attach_disk.go#L52-L56) should happen in both cases.